docs/deploying-identity-provider.md

```diff
@@ -0,0 +1,73 @@
+
+If a service wants to minimize user authentication to reduce friction while signing and has a way to represent a unique user identity, the service could implement a minimally viable OIDC identity provider. The provider would not look like a typical provider that exchanges credentials for a token. The provider could simply generate an identity token for a user that is already authenticated and provide this token to Fulcio (which has allowlisted the provider) or a Sigstore client during artifact signing. The provider would only need to expose the well-known endpoint and verification keys to the internet if the service is signing on behalf of the user. If the user is performing signing in a client rather than through a service, then the token signer would need to be exposed to the user as an authenticated endpoint that exchanges a credential for an identity token.
+
+![Diagram outlining identity provider signing workflow](img/OIDC-provider-sigstore.png)
+
+## Implementation
+
+The service provider needs to implement:
+
+1) A token signer that issues an OIDC compliant ID token for a given identity  
+1) An OIDC provider that conforms to the Discovery specification
+
+A basic example can be found at [https://go.dev/play/p/L2y4Hnvrsek](https://go.dev/play/p/L2y4Hnvrsek), which includes initialization of a token signer, signing a token, initializing OIDC configuration and keys endpoints, verifying and parsing a token, and querying the configuration and keys. An example token for an email identity would be:
+
+```
+{
+  "aud": "sigstore",
+  "email": "test@foo.bar",
+  "email_verified": true,
+  "exp": 1257895800,
+  "iat": 1257894000,
+  "iss": "http://127.0.0.1:36435",
+  "sub": "subject"
+}
+
+```
+
+### Identity
+
+Fulcio has support for many different token identities, though the most common would be [email](https://github.com/sigstore/fulcio/blob/main/docs/oidc.md\#email) and [CI workflow](https://github.com/sigstore/fulcio/blob/main/docs/oid-info.md\#requirements-to-support-signing-with-cicd-workload-identities). Reach out to Fulcio maintainers for more information if you have any questions or additional requirements.
+
+We would expect identities for this type of minimal deployment to be represented as an email. For example, if users authenticate to the service with a username, the email address could be `username@service.com`. If this would conflict with existing email addresses, a service could choose another representation like `username@signing.service.com`. In the case of emails in tokens, Fulcio only verifies that the identity string matches an email address regular expression, not that the email is addressable.
```

Comment:
Per https://github.com/sigstore/fulcio/pull/1741#issuecomment-2248645392: could/should we distinguish between email and non-email identities better here?

Off the top of my head, the current specified behavior is:

* A service can provide an email identity (including a 3p email identity) when that email *is* effectively the user identity on the service. For example, the GitHub user IDP offers `foo@example.com` as a valid email address corresponding to the GitHub account being identified.
* If a service offers non-email user identities, it should use the dedicated `OtherName` form for that service, to avoid confusion/conflation with the email namespace. For example, for a new `widgets.example` IdP, user `foo` should be given the `foo!widgets.example` identifier.